app/cluster/dynamic.go

```diff
@@ -125,6 +163,15 @@ func (d *Dynamic) stop() {
 	d.serverStopCountStat.Increment()
 }
 
+func (d *Dynamic) handleWorkspaceChange(workspaces string) error {
+	if d.currentWorkspaceIDs == workspaces {
+		return nil
+	}
+	d.BackendConfig.StopPolling()
+	d.BackendConfig.StartPolling(workspaces)
+	return nil
```

Comment:
We should also call waitForConfig at this point to ensure the config is completely received by the server